libmattermost.c

```diff
@@ -4098,6 +4319,7 @@ plugin_init(PurplePlugin *plugin)
 	prpl_info->chat_send = mm_chat_send;
 	prpl_info->set_chat_topic = mm_chat_set_topic;
 	prpl_info->add_buddy = mm_add_buddy_no_message;
+	prpl_info->remove_buddy = mm_remove_buddy;

```

Comment:
So there's a bit of a trap with libpurple:

If we add in a buddy-remove function, then when moving a buddy from one group to another, if there's no group-buddy function, then libpurple will 'remove' the buddy from the old group and 'add' the buddy to the new group. 

We'll need to add in a dummy function for grouping buddies so that libpurple doesn't try to remove buddies. eg https://github.com/EionRobb/skype4pidgin/blob/master/skypeweb/libskypeweb.c#L279